### PR TITLE
sshd: Add exec and rework PTY allocation

### PR DESCRIPTION
**Pull request by bluecmd**
exec allows running commands that is not the shell.

For PTYs, let the client control when to allocate a PTY and do not force one
if it elects to not use one. If a PTY is forcefully allocated terminal control
is not user friendly when running for example bash.

### CODE REVIEW

File: cmds/sshd/sshd.go [hunk 1]
```diff
@@ -41,41 +45,34 @@ var (
...
+		if err := p.C.Start(); err != nil {
+			return err
+		}
+		defer p.C.Wait()
+		go io.Copy(p.Ptm, c)
+		go io.Copy(c, p.Ptm)
+	} else {
+		args := strings.Fields(cmd)
+		e := exec.Command(args[0], args[1:]...)
+		e.Stdin = c
```
Comment (rminnich) [inline cmds/sshd/sshd.go]: or e.Stdin, e.Stdout, e.Stderr = c, c, c 
but no big deal

File: cmds/sshd/sshd.go [hunk 2]
```diff
@@ -41,41 +45,34 @@ var (
...
+		e.Stderr = c
+		e.Stdout = c
+		if err := e.Start(); err != nil {
+			return err
+		}
+		log.Printf("Executing non-PTY command %s", cmd)
+		defer e.Wait()
```
Comment (rminnich) [inline cmds/sshd/sshd.go]: Are these defers really worth the effort? Why not just do them in line? I'm not sure I see the point and there's always the issue that you can't totally guarantee when defer happens? I note that in the open source gvisor there are a number of defers that got removed for performance, but in other cases I've seen defer removed due to its lack of precision as to when it gets run.

File: cmds/sshd/sshd.go (hunk 1 above)
Comment (bluecmd) [reply cmds/sshd/sshd.go]: Fixed

File: cmds/sshd/sshd.go (hunk 2 above)
Comment (bluecmd) [reply cmds/sshd/sshd.go]: Fixed, they were leftovers - I just need Process.Wait() and this is going to be moved/changed in a later PR where I add exit code handling.

**Comment by bluecmd** [comment]
Tweaked spawning of exec processes to use user's shell to make u-root behave more like OpenSSH.

PTAL

**Comment by bluecmd** [comment]
Superseeded by #887

**Pull request closed without merging**